src/libponyc/type/cap.c

```diff
@@ -305,26 +288,40 @@ bool is_cap_sub_cap_bound(token_id sub, token_id subalias, token_id super,
       break;
 
     case TK_VAL:
-    case TK_CAP_SHARE:
       switch(super)
       {
         case TK_BOX:
-        case TK_CAP_SHARE:
+        case TK_CAP_SHARE: // {val, tag}
           return true;
 
         default: {}
       }
       break;
 
-    case TK_CAP_READ:
-    case TK_CAP_ALIAS:
+    case TK_CAP_READ: // {iso, trn, ref, val, box}
```

Comment:
Yes on both counts.

Using the rule: "every instantiation of the super rcap must be a supertype of some instantiation of the sub rcap"

- `#share == {val, tag}`
  - `tag` is a supertype of everything, so you can pick any instantiation of `#read` and it's supertype of it
  - `val` is a supertype of itself, which is included in `#read`

- `#alias == {ref, val, box, tag}`
  - `tag` is a supertype of everything, so you can pick any instantiation of `#read` and it's supertype of it
  - `box` is a supertype of `val`, which is included in `#read`
  - `val` is a supertype of itself, which is included in `#read`
  - `ref` is a supertype of itself, which is included in `#read`